src/languages/stan.js

```diff
@@ -523,18 +386,26 @@ export default function(hljs) {
         keywords: 'upper'
       },
       {
-        className: 'keyword',
-        begin: /\btarget\s*\+=/
+        // hack: in range constraints, upper must follow either , or <
+        // <multiplier = ..., offest = ...> or <offset = ...>
+        begin: /[<,]\s*offset\s*=/,
+        keywords: 'offset'
       },
       {
-        begin: '~\\s*(' + hljs.IDENT_RE + ')\\s*\\(',
-        keywords: DISTRIBUTIONS
+        // hack: in range constraints, upper must follow either , or <
+        // <offset = ..., multiplier = ...> or <multiplier = ...>
+        begin: /[<,]\s*multiplier\s*=/,
+        keywords: 'multiplier'
+      },
+      {
+        className: 'keyword',
+        begin: /\btarget\s*/,
       },
       {
         className: 'number',
         variants: [
           {
-            begin: /\b\d+(?:\.\d*)?(?:[eE][+-]?\d+)?/
+            begin: /(?:\b\d+(?:_\d+)*(?:\.(?:\d+(?:_\d+)*)?)?|\B\.\d+(?:_\d+)*)(?:[eE][+-]?\d+(?:_\d+)*)?i?(?!\w)/i
```

Comment:
I see a top-level `|` here do I not?  If so please break this out into multiple variant and add comments on what we're trying to match to aid future maintenance.

Reply:
Shoot, this comes from prism.js and how the author handled complex numbers see https://github.com/PrismJS/prism/blob/c53ad2e65b7193ab4f03a1797506a54bbb33d5a2/components/prism-stan.js#L56. Let me add a reference to that. I definitely don't understand this regex.

Reply:
We prefer our regex split out and properly documented, not just a huge mess of unreadable regex (as found in some other codebases)... if you don't understand this I would revert it and we will wait for someone else to come along and fix it properly.

Reply:
@gibson042 sometimes finds these things fun... not sure if they'd be willing to help here.

Reply:
this does allow for proper handling of complex numbers

Reply:
The question isn't whether it works but whether it's readable and maintainable... our policy generally is we require the different types of numbers to be properly broken out (see many of the other newer grammars, like Swift, etc)... we don't just copy a huge regex from somewhere else because it seems to work (at the current moment).

Reply:
I was adding context for who ever wants to tackle this.

Reply:
I do not think this should be merged unless complex numbers are highlighted appropriately.

Reply:
Entirely disagree and it would be sad to see you stop now since there are a multitude of other improvements we could benefit from.

Reply:
I parsed the regex out into two pieces and commented on them, it's actually a lot simpler than it looks